src/server.c

```diff
@@ -263,80 +264,69 @@ void *epoll_loop(void *ptr) {
 
                 if((in_fd = accept(data->listen_fd, &in_addr, &in_len)) == -1) {
                     if((errno == EAGAIN) || (errno == EWOULDBLOCK)) {
-                        goto epoll_loop_server_reenable;
+                        goto epoll_accept_failed;
                     }
                     else {
                         perror("accept");
-                        goto epoll_loop_server_reenable;
+                        goto epoll_accept_failed;
                     }
                 }
 
                 if(unblock_socket(in_fd) == -1) {
                     //TODO maybe do extra handling of this error case?
                     perror("unblock_socket");
-                    goto epoll_loop_server_reenable;
+                    goto epoll_accept_failed;
                 }
                 if((event.data.ptr = malloc(sizeof(connection_t))) == NULL) {
                     perror("malloc");
-                    goto epoll_loop_server_reenable;
+                    goto epoll_accept_failed;
                 }
 
                 connection_t *new_session = (connection_t *)event.data.ptr;
                 init_connection(new_session, in_fd);
-                event.events = EPOLLIN | EPOLLET | EPOLLONESHOT;
+                event.events = EPOLLIN;
                 if(epoll_ctl(data->epoll_fd, EPOLL_CTL_ADD, in_fd, &event) != 0) {
                     perror("epoll_ctl");
                     free_connection(new_session);
-                    goto epoll_loop_server_reenable;
+                    goto epoll_accept_failed;
                 }
 
-epoll_loop_server_reenable:
-                events[i].events = EPOLLIN | EPOLLET | EPOLLONESHOT;
-                if(epoll_ctl(data->epoll_fd, EPOLL_CTL_MOD, session->fd, &events[i]) != 0) {
-                    perror("epoll_ctl");
-                    continue;
-                }
+epoll_accept_failed: ;
             }
             else {
                 bool done = false;
-                while(1) {
-                    char buf[512];
-                    ssize_t count = -1;
-                    if((count = read(session->fd, buf, sizeof(buf))) == -1) {
-                        if(errno != EAGAIN && errno != EWOULDBLOCK) {
-                            done = true;
-                        }
-                        break;
-                    } else if(count == 0) {
-                        // EOF
+
```

Comment:
Instead of looping through all of this socket's (neverending) data, just take a decent chunk out. Promotes fairness. You can now read the stats page locally instead of the attacker hogging all the network time.